### ISSUE TITLE
The process cannot access the file

### ISSUE THREAD

**Issue by atlithorn** [atom/grunt-electron-installer#53]
Trying to create an electron build. This is the error I get from grunt:
````
Running "create-windows-installer:ia32" (create-windows-installer) task
>> System.IO.IOException: The process cannot access the file 'C:\tmp\installer32
\sling-bridge.0.1.0.nupkg' because it is being used by another process.
>>    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
>>    at System.IO.File.InternalDelete(String path, Boolean checkHost)
>>    at System.IO.File.Delete(String path)
>>    at Squirrel.Update.Program.Releasify(String package, String targetDir, Str
ing packagesDir, String bootstrapperExe, String backgroundGif, String signingOpt
s, String baseUrl, String setupIcon)
>>    at Squirrel.Update.Program.executeCommandLine(String[] args)
>>    at Squirrel.Update.Program.main(String[] args)
>>    at Squirrel.Update.Program.Main(String[] args)
Warning: Command failed: C:\Users\Sling\app\node_modules\grunt-electron-installe
r\vendor\Update.com --releasify C:\Users\Sling\AppData\Local\Temp\squirrel-insta
ller-115727-2100-1hk65vl\sling-bridge.0.1.0.nupkg --releaseDir C:\tmp\installer3
2 --loadingGif C:\Users\Sling\app\node_modules\grunt-electron-installer\resource
s\install-spinner.gif
System.IO.IOException: The process cannot access the file 'C:\tmp\installer32\sl
ing-bridge.0.1.0.nupkg' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalDelete(String path, Boolean checkHost)
   at System.IO.File.Delete(String path)
   at Squirrel.Update.Program.Releasify(String package, String targetDir, String
 packagesDir, String bootstrapperExe, String backgroundGif, String signingOpts,
String baseUrl, String setupIcon)
   at Squirrel.Update.Program.executeCommandLine(String[] args)
   at Squirrel.Update.Program.main(String[] args)
   at Squirrel.Update.Program.Main(String[] args)
 Use --force to continue.

Aborted due to warnings.

```
I have double and triple checked and there is no process running that could be the cause of this (at least not before running grunt)

Any ideas?

**Comment by paulcbetts** [comment]
Dunno!

**Comment by eightytwo** [comment]
I'm experiencing the same problem. For what it's worth I'm running this in a virtual machine so I'm not sure if that's causing issues. Procmon reports an access violation but I can't see any other process with a lock on the nupkg file.

I'm pretty sure it's this [`File.Delete`](https://github.com/Squirrel/Squirrel.Windows/blob/e5a067dfd3b852269d2636b37d462213b3a0473c/src/Update/Program.cs#L369) that isn't able to execute successfully. If I run build and run Update.exe from the Squirrel.Windows source with the above line commented out the process runs fine.

Hope this extra information helps.

**Comment by gbmhunter** [comment]
Sorry, I had made a duplicate, so I closed it and will post my issue here (essentially the same thing).

**Comment by gbmhunter** [comment]
Upon running "grunt create-windows-installer" I get the following error:

Running "create-windows-installer:x64" (create-windows-installer) task
>> System.IO.IOException: The process cannot access the file 'C:\temp\output\Nin
jaTerm.0.1.0.nupkg' because it is being used by another process.
>>    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
>>    at System.IO.File.InternalDelete(String path, Boolean checkHost)
>>    at System.IO.File.Delete(String path)
>>    at Squirrel.Update.Program.Releasify(String package, String targetDir, Str
ing packagesDir, String bootstrapperExe, String backgroundGif, String signingOpt
s, String baseUrl, String setupIcon)
>>    at Squirrel.Update.Program.executeCommandLine(String[] args)
>>    at Squirrel.Update.Program.main(String[] args)
>>    at Squirrel.Update.Program.Main(String[] args)
I have made sure the output directory is not in any cloud-syncing folder, and even run the command-line in administrator mode but no luck. I have tried putting the output file somewhere completely different on my computer but that didn't work either.

I am running Windows 8 64-bit.

**Comment by gbmhunter** [comment]
There is the line (currently line 31):

```
done = this.async();
``` 

in node_modules/grunt-electron-installer/tasks/index.js which indicates that grunt tasks are being run asynchronously. I commented this line out in attempt to make them run one at a time (albeit slower), and this got rid of the error. However, no .exe file was generated in the output directory.

**Comment by gbmhunter** [comment]
@eightytwo are you able to elaborate on how you got it to work? Do I have to download the squirrel source code and use that rather than what is in node_modules/grunt-electron-installer/vendor?

**Comment by eightytwo** [comment]
@gbmhunter, exactly. I cloned the [Squirrel.Windows](https://github.com/Squirrel/Squirrel.Windows) repository and commented out [this](https://github.com/Squirrel/Squirrel.Windows/blob/e5a067dfd3b852269d2636b37d462213b3a0473c/src/Update/Program.cs#L369) line (which was the file operation that was failing) and built a new version of Update.exe. I then copied this executable into the `node_modules/grunt-electron-installer/vendor` directory and ran the grunt task. While this worked I haven't looked any further into the file lock issue.

**Comment by thughes** [comment]
I've occasionally hit this as well. Once it starts happening, it seems to repeatedly happen. I restarted my Windows 10 VM and cloned everything into a new directory and it started working again.

**Comment by gbmhunter** [comment]
@eightytwo's suggestion about modifying the `Update.exe` worked for me also (after having to deal with some issues in building Squirrel). If you want a modified, pre-compiled `Update.exe` (which might work on your machine), I have uploaded it to https://github.com/mbedded-ninja/NinjaTerm , under the misc/ folder.

This `Update.exe` needs to replace the one in `node_modules/grunt-electron-installer/vendor`.

**Comment by paulcbetts** [comment]
I think this is a bug in Squirrel that we've recently fixed where we leaked a handle. We'll get a new version out this week.

**Comment by gbmhunter** [comment]
Great stuff.

**Comment by paulcbetts** [comment]
This should be fixed in 1.0.1 or higher

**Comment by gbmhunter** [comment]
I can confirm v1.0.2 fixes the bug for me. Thanks!
